OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -269,37 +269,41 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 							return true;
 						}
 
-						if (!ValidateSlotCommand(server, conn, client, parts[0], true))
+						var slotKey = parts[0];
+
+						if (!ValidateSlotCommand(server, conn, client, slotKey, true))
 							return false;
 
-						var slot = server.LobbyInfo.Slots[parts[0]];
-						var bot = server.LobbyInfo.ClientInSlot(parts[0]);
+						var slot = server.LobbyInfo.Slots[slotKey];
+						var bot = server.LobbyInfo.ClientInSlot(slotKey);
 						int controllerClientIndex;
 						if (!Exts.TryParseIntegerInvariant(parts[1], out controllerClientIndex))
 						{
 							Log.Write("server", "Invalid bot controller client index: {0}", parts[1]);
 							return false;
 						}
 
-						var botType = parts.Skip(2).JoinWith(" ");
-
 						// Invalid slot
 						if (bot != null && bot.Bot == null)
 						{
 							server.SendOrderTo(conn, "Message", "Can't add bots to a slot with another client.");
 							return true;
 						}
 
+						var botType = parts.Skip(2).JoinWith(" ");
+						var slotIdx = server.LobbyInfo.Slots.Keys.ToArray().IndexOf(slotKey);
+						var botName = "{0} {1}".F(botType, slotIdx + 1);
```

Comment:
I agree, the reason I used slot number and not number of bots is that while in lobby, I would need to renumber all following bots and I wanted to avoid this to keep it simple. Do I understand you correctly you propose to name bots after game start? (or also in lobby?)